crates/matrix-sdk/src/client/builder.rs

```diff
@@ -471,12 +472,19 @@ impl ClientBuilder {
         let mut sliding_sync_proxy =
             self.sliding_sync_proxy.as_ref().map(|url| Url::parse(url)).transpose()?;
 
-        #[allow(unused_variables)]
-        if let Some(well_known) = well_known {
-            #[cfg(feature = "experimental-sliding-sync")]
-            if sliding_sync_proxy.is_none() {
-                sliding_sync_proxy =
-                    well_known.sliding_sync_proxy.and_then(|p| Url::parse(&p.url).ok())
+        #[cfg(feature = "experimental-sliding-sync")]
+        if self.is_simplified_sliding_sync_enabled {
+            // When using Simplified MSC3575, don't use a sliding sync proxy, allow the
+            // requests to be sent directly to the homeserver.
+            tracing::info!("Simplified MSC3575 is enabled, ignoring any sliding sync proxy.");
+            sliding_sync_proxy = None;
+        } else {
+            // Otherwise, if a proxy wasn't set, use the one discovered from the well-known.
+            if let Some(well_known) = well_known {
```

Comment:
```suggestion
        } else if let Some(well_known) = well_known {
            // Otherwise, if a proxy wasn't set, use the one discovered from the well-known.
```